lib/remote/apilistener-configsync.cpp

```diff
@@ -28,12 +28,17 @@ void ApiListener::ConfigUpdateObjectHandler(const ConfigObject::Ptr& object, con
 	if (!listener)
 		return;
 
-	if (object->IsActive()) {
-		/* Sync object config */
-		listener->UpdateConfigObject(object, cookie);
-	} else if (!object->IsActive() && object->GetExtension("ConfigObjectDeleted")) {
-		/* Delete object */
-		listener->DeleteConfigObject(object, cookie);
+	try {
+		if (object->IsActive()) {
+			/* Sync object config */
+			listener->UpdateConfigObject(object, cookie);
+		} else if (!object->IsActive() && object->GetExtension("ConfigObjectDeleted")) {
+			/* Delete object */
+			listener->DeleteConfigObject(object, cookie);
+		}
+	} catch (const std::runtime_error& ex) {
+		Log(LogInformation, "ApiListener")
+			<< "Error while processing config update message: " << DiagnosticInformation(ex, false);
 	}
```

Comment:
You are right! I don't even need to catch any errors here, since `client` is not provided and there would be no unhandled exceptions. The two methods would only throw an error if the client param is set, which is not the case here.